packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss

```diff
@@ -148,7 +148,7 @@
 
 // Add space between adjacent MessageBars.
 .root + .root {
-  margin-bottom: 6px;
+  margin-top: 6px;
```

Comment:
I was being clever creating a PR from within github interface :)

As for positioning approach, I agree that a container is better for the future, but this is fixing the immediate bug that margin has effect on everything EXCEPT the first element. 

I'll whip up a change file, but I still feel this needs to go in now, and we can talk about a layoutSet later (or sooner)